src/Arcus.Security.Providers.KeyPerFile/KeyPerFileSecretProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration.KeyPerFile;
+using System;
+using System.Threading.Tasks;
+
+namespace Arcus.Security.Providers.KeyPerFile
+{
+    public class KeyPerFileSecretProvider : ISecretProvider
+    {
+        private readonly KeyPerFileConfigurationProvider _provider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="KeyPerFileSecretProvider"/> class.
+        /// </summary>
```

Comment:
```suggestion
        /// <summary>
        /// Initializes a new instance of the <see cref="KeyPerFileSecretProvider"/> class.
        /// </summary>
        /// <param name="configurationSource">The configuration source that provides the Docker secrets mounted as files in the container.</param>
        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="configurationSource"/> is <c>null</c></exception>
```

Provides some argument information.

Reply:
Done.   I did it via VS.NET instead of committing the suggestion, since I've renamed some files and directories.